catkin_ws/src/planner/src/substates/utility/controller.py

```diff
@@ -109,6 +114,26 @@ def get_state_goal(self,state,keepers,displace):
         goal.displace = displace
         goal.do_x, goal.do_y, goal.do_z, goal.do_theta_x, goal.do_theta_y, goal.do_theta_z = keepers
         return goal
+    
+    def pose_goal(self,position,quaternion,keepers,displace):
+        x,y,z = position
+        w,qx,qy,qz = quaternion
+        goal = QuaternionGoal()
+        goal.pose.position.x = x
+        goal.pose.position.y = y
+        goal.pose.position.z = z
+        goal.pose.orientation.w = w
+        goal.pose.orientation.x = qx
+        goal.pose.orientation.y = qy
+        goal.pose.orientation.z = qz
+        goal.displace = displace
+        goal.do_x, goal.do_y, goal.do_z, goal.do_quaternion = keepers
+        return goal
+    
+    def quaternion_action(self,position,quaternion):
```

Comment:
unnecesary function mentioned above

Reply:
Yes because right now we are only implementing code for experimenting with quaternions. Switching the implementation of move and rotate would be too much right now